src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/SpotEllipsoidFeatureTest.java

```diff
@@ -0,0 +1,102 @@
+			assertEquals( expectedShortAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.SHORT_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedMiddleAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.MIDDLE_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedLongAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.LONG_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			// volume of ellipsoid https://en.wikipedia.org/wiki/Ellipsoid#Volume
+			assertEquals( expectedVolume, ellipsoidFeature
+					.project( FeatureProjectionKey.key( SpotEllipsoidFeature.VOLUME_PROJECTION_SPEC ) ).value( exampleGraph3.spot0 ),
+					0.01d );
+
+			Model model = exampleGraph3.getModel();
+			FeatureSerializerTestUtils.saveAndReload( context, exampleGraph3.getModel(), ellipsoidFeature );
+
+			// check that the feature has correct values after saving and reloading
+			SpotEllipsoidFeature ellipsoidFeatureReloaded = ( SpotEllipsoidFeature ) model.getFeatureModel()
+					.getFeature( SpotEllipsoidFeature.SPOT_ELLIPSOID_FEATURE_SPEC );
+			assertEquals( expectedShortAxis,
+					ellipsoidFeatureReloaded.project( FeatureProjectionKey.key( SpotEllipsoidFeature.SHORT_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedMiddleAxis, ellipsoidFeatureReloaded
+					.project( FeatureProjectionKey.key( SpotEllipsoidFeature.MIDDLE_SEMI_AXIS_PROJECTION_SPEC ) )
+					.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedLongAxis,
+					ellipsoidFeatureReloaded.project( FeatureProjectionKey.key( SpotEllipsoidFeature.LONG_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedVolume,
+					ellipsoidFeatureReloaded.project( FeatureProjectionKey.key( SpotEllipsoidFeature.VOLUME_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.01d );
+
+			// test, if features are NaN after invalidation
+			ellipsoidFeature.invalidate( exampleGraph3.spot0 );
+			assertTrue( Double.isNaN(
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.SHORT_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ) ) );
+			assertTrue( Double.isNaN(
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.MIDDLE_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ) ) );
+			assertTrue( Double.isNaN(
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.LONG_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ) ) );
+			assertTrue( Double.isNaN(
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.VOLUME_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ) ) );
+
```

Comment:
According to clean code, each test should test only one thing. This test tests feature computation, feature serialization and invalidation. Could split it up without making it more complicated? (Just a suggestion)

Reply:
Yes, you are right. I was actually aware of this, but did not see, how to separate the tests, without making it more complicated. I think, I found a solution now.